ext/pdo/pdo_stmt.c

```diff
@@ -938,6 +939,7 @@ static bool do_fetch(pdo_stmt_t *stmt, zval *return_value, enum pdo_fetch_type h
 		}
 		efree(fetch_function_params);
 	}
+	stmt->in_fetch = false;
```

Comment:
Not all exit paths in this function set `in_fetch` back to false. You probably need an `out`/`err` label; or you need to put this `in_fetch` setting at the callsite